lib/corebulk.go

```diff
@@ -306,9 +312,18 @@ func (b *BulkIndexer) Delete(index, _type, id string) {
 	return
 }
 
-func (b *BulkIndexer) UpdateWithWithScript(index string, _type string, id, parent, ttl string, date *time.Time, script string) error {
+func (b *BulkIndexer) UpdateWithWithScript(index string, _type string, id, parent, ttl string, date *time.Time, script EsScript, scripted_upsert bool, upsert interface{}) error {
 
 	var data map[string]interface{} = make(map[string]interface{})
+	if scripted_upsert {
+		data["scripted_upsert"] = true
+	}
+	if upsert != nil {
+		data["upsert"] = upsert
+	} else {
+		data["upsert"] = map[string]interface{}{}
```

Comment:
Does this need to be set in all cases?

Reply:
No, that was suppose to be inside the scripted_upsert (if that is set to true, you must have an upsert key)